src/main/scala/org/constellation/primitives/storage/CheckpointService.scala

```diff
@@ -54,6 +55,33 @@ class CheckpointBlocksMemPool(size: Int = 50000)(implicit dao: DAO)
 
 object CheckpointService {
   def apply(implicit dao: DAO, size: Int = 50000) = new CheckpointService(dao, size)
+
+  def fetchFullData(cacheData: CheckpointCacheData)(implicit dao: DAO): CheckpointCacheFullData = {
+    CheckpointCacheFullData(
+      Some(
+        CheckpointBlockFullData(
+          fetchTransactions(cacheData.cb.transactionsMerkleRoot),
+          cacheData.cb.checkpoint,
+          fetchMessages(cacheData.cb.messagesMerkleRoot),
+          fetchNotifications(cacheData.cb.notificationsMerkleRoot)
+        )
+      ),
+      cacheData.children,
+      Some(cacheData.height)
+    )
+  }
```

Comment:
@ryle-goehausen shall we introduce new storage pool for resolving `rootHash => Seq[Data]` ?